### PR TITLE
docs(cli): add examples for `argo archive list` with option explanations

### PR DESCRIPTION
**Pull request by leehosu**
Fixes #11898 

### Motivation

Modify the `argo archive list` command for more information

### Modifications

Added examples for the argo archive list command.

### Verification

run `argo archive list -h`

### CODE REVIEW

File: v3
Comment (agilgur5) [inline v3:1]: This shouldn't be committed

File: cmd/argo/commands/archive/list.go [hunk 1]
```diff
@@ -25,6 +25,21 @@ func NewListCommand() *cobra.Command {
...
+
+# Filter list based on a specific label selector :
+argo archive list -l key1=value1,key2=value2
+
+# Fetch the list in chunks instead of all at once :
+argo archive list --chunk-size 100
+
+# Display help information for the command :
+argo archive list -h
+`,
```
Comment (agilgur5) [inline cmd/argo/commands/archive/list.go:42]: Per the issue description: "See already-checked items for inspiration, use similar style & examples when possible, such as `argo list` for `argo archive list`, etc." -- this should be similar to https://github.com/argoproj/argo-workflows/pull/12076

File: docs/cli/argo_archive_list.md [hunk 2]
```diff
@@ -6,6 +6,25 @@ list workflows in the archive
...
+
+# Filter list based on a specific label selector
+argo archive list -l key1=value1,key2=value2
+
+# Fetch the list in chunks instead of all at once
+argo archive list --chunk-size 100
+
+# Display help information for the command
+argo archive list -h
+```
```
Comment (agilgur5) [inline docs/cli/argo_archive_list.md:26]: This is failing CI Codegen as it must match the CLI code verbatim

**Review by agilgur5 — CHANGES_REQUESTED** [review]
Thanks for the contribution. Please try to follow the issue description and it's recommendations closely, see in-line comments below

> ### Verification
> Added examples for the argo archive list command.

Verification means checking that these examples work and that the changes are tested / pass CI

File: docs/cli/argo_archive_list.md (hunk 2 above)
Comment (leehosu) [reply docs/cli/argo_archive_list.md:23]: @agilgur5 I don't know why I keep failing the CI test. Can you tell me a little more?

Comment (MasonM) [reply docs/cli/argo_archive_list.md:24]: @leehosu That error should hopefully be fixed with https://github.com/argoproj/argo-workflows/pull/13660. After that's merged, click the "Update branch" button at the bottom of the PR

Comment (leehosu) [reply docs/cli/argo_archive_list.md:24]: @MasonM Thank you for your help! I hope it's merge...

File: cmd/argo/commands/archive/list.go [hunk 3]
```diff
@@ -25,6 +25,18 @@ func NewListCommand() *cobra.Command {
...
+  argo archive list
+
+# List all archives in JSON format:
+  argo archive list -o json
+
+# List archives filtered that have both labels:
+  argo archive list -l key1=value1,key2=value2
+
+# List archives fetched in chunks of 100:
+  argo archive list --chunk-size 100
```
Comment (agilgur5) [inline cmd/argo/commands/archive/list.go:38]: The subject is not "archives" and there is only one "archive". The term you're looking for might be "archived workflows"

```suggestion
		Example: `# List all archived workflows:
  argo archive list

# List all archived workflows fetched in chunks of 100:
  argo archive list --chunk-size 100

# List all archived workflows in YAML format:
  argo archive list -o yaml

# List archived workflows that have both labels:
  argo archive list -l key1=value1,key2=value2
```

- Regarding the label examples, match the wording from #12706
  - use `-o YAML` same as #12706 as a more common format as well
- move the chunks to the 2nd example, as they may be quite useful

**Review by agilgur5 — COMMENTED** [review]
Some grammar and style changes below. Also you have not yet corrected your verification section, per my previous comment

**Comment by agilgur5** [comment]
Codegen is [failing](https://github.com/argoproj/argo-workflows/actions/runs/11082290195/job/30795090086?pr=13662) correctly, you'll need to apply my [previous suggestion](https://github.com/argoproj/argo-workflows/pull/13662#discussion_r1777738542) to the generated CLI docs as well.

I did not mention that initially as that was intended as a learning opportunity for you to figure out on your own. As you've merged `main` twice for CI, it seems you might have missed that.

**Review by agilgur5 — APPROVED** [review]
LGTM now, thanks

**Pull request merged**